dart/constraint/ConstrainedGroup.h

```diff
@@ -107,7 +108,7 @@ class ConstrainedGroup
   std::vector<ConstraintBase*> mConstraints;
 
   ///
-  dynamics::Skeleton* mRootSkeleton;
+  std::shared_ptr<dynamics::Skeleton> mRootSkeleton;
```

Comment:
`DynamicsTypes.h`  is just my preference rather than create different type defining files like `InvalidIndex.h` if it doesn't cause too much coupling. But I'm fine with `Ptr.h` and `InvalidIndex.h`.

Beside on that, how do you think about of using const variable instead enum?